controller/Previewer.php

```diff
@@ -43,68 +63,66 @@ public function __construct()
     }
 
     /**
-     * Gets an error response object
-     * @param Exception [$e] Optional exception from which extract the error context
-     * @param array $prevResponse Response before catch
+     * Gets an error response array
+     * @param Exception $e
      * @return array
      */
-    protected function getErrorResponse($e = null)
+    protected function getErrorResponse($e)
     {
         $response = [
             'success' => false,
             'type' => 'error',
         ];
 
-        if ($e) {
-            if ($e instanceof \Exception) {
-                $response['type'] = 'exception';
-                $response['code'] = $e->getCode();
-            }
+        switch (true) {
+            case $e instanceof FileNotFoundException:
+                $response['type'] = 'FileNotFound';
+                $response['message'] = __('File not found');
+                break;
 
-            if ($e instanceof \common_exception_UserReadableException) {
+            case $e instanceof UnauthorizedException:
+                $response['code'] = 403;
                 $response['message'] = $e->getUserMessage();
-            } else {
+                break;
+            case $e instanceof UserReadableException:
+                $response['message'] = $e->getUserMessage();
+                break;
+            case $e instanceof Exception:
+                $response['type'] = 'exception';
+                $response['code'] = $e->getCode();
+                $response['message'] = $e->getMessage();
+                break;
+            default:
                 $response['message'] = __('An error occurred!');
-            }
-
-            switch (true) {
-                case $e instanceof \tao_models_classes_FileNotFoundException:
-                    $response['type'] = 'FileNotFound';
-                    $response['message'] = __('File not found');
-                    break;
-
-                case $e instanceof \common_exception_Unauthorized:
-                    $response['code'] = 403;
-                    break;
-            }
+                break;
         }
 
         return $response;
     }
 
     /**
      * Gets an HTTP response code
-     * @param Exception [$e] Optional exception from which extract the error context
+     * @param Exception $e
      * @return int
      */
-    protected function getErrorCode($e = null)
+    protected function getErrorCode($e)
     {
-        $code = 200;
```

Comment:
please verify usage, because the removed 200 might have been a expected code for other side of the codebase

Reply:
I cannot find any usages of that code... And I don't see a reason to generate an error with exception and with 200